packages/tangle/message.go

```diff
@@ -344,90 +408,93 @@ func MessageFromBytes(bytes []byte) (result *Message, consumedBytes int, err err
-			err = errors.Errorf("failed to parse parents count from MarshalUtil: %w", err)
-			return
+			return nil, errors.Errorf("failed to parse parents count from MarshalUtil: %w", err)
 		}
 		references := make(MessageIDs, parentsCount)
 		for j := 0; j < int(parentsCount); j++ {
 			if references[j], err = ReferenceFromMarshalUtil(marshalUtil); err != nil {
-				err = errors.Errorf("failed to parse parent %d-%d from MarshalUtil: %w", i, j, err)
-				return
+				return nil, errors.Errorf("failed to parse parent %d-%d from MarshalUtil: %w", i, j, err)
 			}
 		}
-		result.parentsBlocks[i] = ParentsBlock{
+		parentsBlocks[i] = ParentsBlock{
 			ParentsType:  ParentsType(parentType),
 			ParentsCount: parentsCount,
 			References:   references,
 		}
 	}
 
-	if result.issuerPublicKey, err = ed25519.ParsePublicKey(marshalUtil); err != nil {
-		err = fmt.Errorf("failed to parse issuer public key of the message: %w", err)
-		return
+	issuerPublicKey, err := ed25519.ParsePublicKey(marshalUtil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse issuer public key of the message: %w", err)
 	}
-	if result.issuingTime, err = marshalUtil.ReadTime(); err != nil {
-		err = fmt.Errorf("failed to parse issuing time of the message: %w", err)
-		return
+	issuingTime, err := marshalUtil.ReadTime()
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse issuing time of the message: %w", err)
 	}
-	if result.sequenceNumber, err = marshalUtil.ReadUint64(); err != nil {
-		err = fmt.Errorf("failed to parse sequence number of the message: %w", err)
-		return
+	msgSequenceNumber, err := marshalUtil.ReadUint64()
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse sequence number of the message: %w", err)
 	}
-	if result.payload, err = payload.FromMarshalUtil(marshalUtil); err != nil {
-		err = fmt.Errorf("failed to parse payload of the message: %w", err)
-		return
+
+	msgPayload, err := payload.FromMarshalUtil(marshalUtil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse payload of the message: %w", err)
 	}
-	if result.nonce, err = marshalUtil.ReadUint64(); err != nil {
-		err = fmt.Errorf("failed to parse nonce of the message: %w", err)
-		return
+
+	nonce, err := marshalUtil.ReadUint64()
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse nonce of the message: %w", err)
 	}
-	if result.signature, err = ed25519.ParseSignature(marshalUtil); err != nil {
-		err = fmt.Errorf("failed to parse signature of the message: %w", err)
-		return
+	signature, err := ed25519.ParseSignature(marshalUtil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse signature of the message: %w", err)
 	}
 
 	// retrieve the number of bytes we processed
 	readOffsetEnd := marshalUtil.ReadOffset()
 
 	// store marshaled version as a copy
-	result.bytes, err = marshalUtil.ReadBytes(readOffsetEnd-readOffsetStart, readOffsetStart)
+	msgBytes, err := marshalUtil.ReadBytes(readOffsetEnd-readOffsetStart, readOffsetStart)
```

Comment:
Do we check that there are no bytes left to process at this point?

Reply:
I didn't add it..

But if you run the test that checks just this on line 890 of message_test it passes just fine

Reply:
As far as I understand line 870 of message_test is not the same test: in that you are checking the the amount of bytes processed equals the amount of bytes in the message. I would like the parser to fail if the message has unprocessed trailing bytes, with a corresponding test.
I think it will make the parser more robust! What do you think?

Reply:
I didn't forget, was just getting delayed...

Currently there is the following code:
```
// retrieve the number of bytes we processed
	readOffsetEnd := marshalUtil.ReadOffset()

	// store marshaled version as a copy
	msgBytes, err := marshalUtil.ReadBytes(readOffsetEnd-readOffsetStart, readOffsetStart)
	if err != nil {
		return nil, fmt.Errorf("error trying to copy raw source bytes: %w", err)
	}
```

It makes sure that we only consider the bytes that we read as the real bytes, meaning if there are more bytes in the buffer we simply trim them and never use them.
I think this works.. I wouldn't have doen it this way, but not sure if I should make more changes to this PR, especially since we will be soon have a PR rewriting the deserializer...